pkg/policy/identity.go

```diff
@@ -63,21 +65,82 @@ func (id NumericIdentity) Uint32() uint32 {
 	return uint32(id)
 }
 
-type RuleContexts map[RuleContext]bool
+// SecurityIDContexts maps a security identity to a L4RuleContexts
+type SecurityIDContexts map[NumericIdentity]L4RuleContexts
 
-// RuleContext represents a L3-dependent L4 rule
+func (sc SecurityIDContexts) DeepCopy() SecurityIDContexts {
+	cpy := make(SecurityIDContexts)
+	for k, v := range sc {
+		cpy[k] = v.DeepCopy()
+	}
+	return cpy
+}
+
+// SecurityIDContexts returns a new L4RuleContexts created.
+func NewSecurityIDContexts() SecurityIDContexts {
+	return SecurityIDContexts(make(map[NumericIdentity]L4RuleContexts))
+}
+
+// L4RuleContexts maps a rule context to a boolean.
+type L4RuleContexts map[L4RuleContext]L7RuleContext
```

Comment:
What does this boolean represent?

Reply:
It's not even a boolean! :P

Reply:
It was just to see if you were paying attention. Fixed.